config/schemas/operator.schema

```diff
@@ -0,0 +1,219 @@
+      "items": {
+        "$schema": "http://json-schema.org/draft-04/schema#",
+        "$ref": "#/definitions/OperatorDependency"
+      },
+      "type": "array"
+    },
+    "description": {
+      "type": "string"
+    },
+    "kubernetesVersion": {
+      "type": "string",
+      "description": "min version of Kubernetes required for this operator which must be semver x.y.z",
+      "examples": [
+        "0.1.1"
+      ]
+    },
+    "kudoVersion": {
+      "type": "string",
+      "description": "min version of KUDO required for this operator which must be semver",
+      "examples": [
+        "0.7.0"
+      ]
+    },
+    "maintainers": {
+      "items": {
+        "$schema": "http://json-schema.org/draft-04/schema#",
+        "$ref": "#/definitions/Maintainer"
+      },
+      "type": "array",
+      "description": "list of maintainers"
+    },
+    "name": {
+      "maxLength": 253,
+      "minLength": 1,
+      "type": "string",
+      "description": "The name of the operator which must be less than 254 chars and be lower case alphanumeric and/or . and -"
+    },
+    "plans": {
+      "patternProperties": {
+        ".*": {
+          "$schema": "http://json-schema.org/draft-04/schema#",
+          "$ref": "#/definitions/Plan"
+        }
+      },
+      "type": "object",
+      "description": "a collection of Plans by name"
+    },
+    "tasks": {
+      "patternProperties": {
+        ".*": {
+          "$schema": "http://json-schema.org/draft-04/schema#",
+          "$ref": "#/definitions/TaskSpec"
+        }
+      },
+      "type": "object",
+      "description": "a collection of Tasks to be used in the Steps of Plans mapped by a name"
+    },
+    "url": {
+      "type": "string",
+      "description": "project URL for the operator",
+      "examples": [
+        "http://kudo.dev"
+      ]
+    },
+    "version": {
+      "type": "string",
+      "description": "version of the operator must be semver x.y.z",
+      "examples": [
+        "0.1.1"
+      ]
+    }
+  },
+  "additionalProperties": false,
+  "type": "object",
+  "definitions": {
+    "Maintainer": {
+      "properties": {
+        "email": {
+          "type": "string",
+          "description": "email for maintainer",
+          "examples": [
+            "billy@kudo.dev"
+          ]
+        },
+        "name": {
+          "type": "string",
+          "description": "name of maintainer",
+          "examples": [
+            "Billy Bob"
+          ]
+        }
+      },
+      "additionalProperties": false,
+      "type": "object"
+    },
+    "OperatorDependency": {
+      "additionalProperties": true,
+      "type": "object"
+    },
+    "Phase": {
+      "required": [
+        "name",
+        "strategy",
+        "steps"
+      ],
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "name of the phase",
+          "examples": [
+            "deploy-zookeeper"
+          ]
+        },
+        "steps": {
```

Comment:
Will there be an issue with having this lowercase but `Phase` uppercase ? Should all be lowercase?

Reply:
it should be all lowercase :(   the tooling doesn't support that (without changes).   It uses go reflect fo the definition of type names.

Reply:
it works the way it is... this will be another point of decision for the team.